src/utils_request.js

```diff
@@ -0,0 +1,272 @@
+ * hasn't been sent by a full web browser. Using a headless browser for such requests
+ * is an order of magnitude more resource-intensive than this function.
+ *
+ * Currently, the function sends requests the same way as Firefox web browser does.
+ * In the future, it might add support for other browsers too.
+ *
+ * Internally, the function uses `requestBetter()` function to perform the request.
+ * All `options` not recognized by this function are passed to it,
+ * so see it for more details.
+ *
+ * @param options.url
+ *  URL of the target endpoint. Supports both HTTP and HTTPS schemes.
+ * @param [options.method=GET]
+ *  HTTP method.
+ * @param [options.headers={}]
+ *  Additional HTTP headers to add. It's recommended not to use this option,
+ *  because it can ruin the signature of the web browser. TODO: Maybe let's remove this completely?
+ * @param [options.languageCode=en]
+ *  Two-letter ISO 639 language code.
+ * @param [options.countryCode=US]
+ *  Two-letter ISO 3166 country code.
+ * @param [options.isMobile]
+ *  If `true`, the function uses User-Agent of a mobile browser.
+ *
+ * @return {{ response, body }}
+ *  Returns an object with two properties: `response` is the instance of
+ *  Node's [`http.IncomingMessage`](https://nodejs.org/api/http.html#http_class_http_incomingmessage) class,
+ *  `body` is a `String`, `Buffer` or `Object`, depending on the `encoding` and `json` options.
+ */
+export const requestLikeBrowser = async (options) => {
+    const defaultOptions = {
+        countryCode: 'US',
+        languageCode: 'en',
+        headers: {},
+        method: 'GET',
+        useMobileVersion: false,
+    };
+    const opts = Object.assign(defaultOptions, options);
+
+    const parsedUrl = url.parse(opts.url);
+
+    const browserHeaders = {
+        Host: parsedUrl.host,
+        'User-Agent': opts.useMobileVersion ? FIREFOX_MOBILE_USER_AGENT : FIREFOX_DESKTOP_USER_AGENT,
+        Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
+        'Accept-Language': `${opts.languageCode}-${opts.countryCode},${opts.languageCode};q=0.5`,
+        'Accept-Encoding': 'gzip, deflate, br',
+        DNT: '1',
+        Connection: 'keep-alive',
+        'Upgrade-Insecure-Requests': '1',
+    };
+    opts.headers = Object.assign(browserHeaders, opts.headers);
+
+    const response = await requestBetter(opts);
+    const { type } = contentType.parse(response.headers['content-type']);
+
+    // Handle situations where the server explicitly states that
+    // it will not serve the resource as text/html by skipping.
+    if (response.statusCode === 406) {
+        throw new Error(`requestLikeBrowser: Resource ${options.url} is not available in HTML format. Skipping resource.`);
+    }
+
+    if (type.toLowerCase() !== 'text/html') {
+        throw new Error(`Received unexpected Content-Type: ${type}`);
+    }
+
+    const { body } = response;
+
+    if (!body) throw new Error('The response body is empty');
```

Comment:
Not sure that we should throw on empty string. Does Request throw?

Reply:
Actually, I 100% positive it does not, because no body could be the 204 response code. However, in this context, I think that the `requestLikeBrowser` wants to get some kind of body since we are checking whether the `content-type` is HTML.

Reply:
Tried it in browser and it just shows an empty screen. No error in console.

Reply:
That is true. I wanted to make a different point here. The purpose of the `requestLikeBrowser` is to get HTML for the page, so I would consider returning everything except the HTML as a bug. If we would be returning the empty or no body, this would cause having a check for a no body everywhere the function is used in my opinion.

Reply:
Empty content with a 200-OK is still content when you're scraping. The crawlers should not retry and then mark the request as failed just because some page had empty HTML I think.